commons/src/main/kotlin/org/fossify/commons/dialogs/PropertiesDialog.kt

```diff
@@ -189,6 +189,42 @@ class PropertiesDialog : BasePropertiesDialog {
                         }
                     }
                 }
+
+                addProperty(R.string.sha1, "…", R.id.properties_sha1)
+                ensureBackgroundThread {
+                    val sha1 = if (mActivity.isRestrictedSAFOnlyRoot(path)) {
+                        mActivity.contentResolver.openInputStream(mActivity.getAndroidSAFUri(path))?.sha1()
+                    } else {
+                        File(path).sha1()
+                    }
+
+                    mActivity.runOnUiThread {
+                        if (sha1 != null) {
+                            (mDialogView.propertiesHolder.findViewById<LinearLayout>(R.id.properties_sha1).findViewById<MyTextView>(R.id.property_value)).text =
+                                sha1
+                        } else {
+                            mDialogView.propertiesHolder.findViewById<LinearLayout>(R.id.properties_sha1).beGone()
+                        }
+                    }
+                }
+
+                addProperty(R.string.sha256, "…", R.id.properties_sha256)
+                ensureBackgroundThread {
+                    val sha256 = if (mActivity.isRestrictedSAFOnlyRoot(path)) {
+                        mActivity.contentResolver.openInputStream(mActivity.getAndroidSAFUri(path))?.sha256()
+                    } else {
+                        File(path).sha256()
+                    }
+
+                    mActivity.runOnUiThread {
+                        if (sha256 != null) {
+                            (mDialogView.propertiesHolder.findViewById<LinearLayout>(R.id.properties_sha256).findViewById<MyTextView>(R.id.property_value)).text =
+                                sha256
+                        } else {
+                            mDialogView.propertiesHolder.findViewById<LinearLayout>(R.id.properties_sha256).beGone()
+                        }
+                    }
+                }
```

Comment:
That's a lot of repetition. The common logic should be extracted to a function.

You could do something like the following:

```kotlin
private fun calculateAndDisplayHash(
    labelRes: Int, // e.g. R.string.md5 or R.string.sha1
    propertyId: Int, // e.g. R.id.properties_md5  or R.string.sha256
    hashInputStream: (InputStream) -> String?, // Pass { inputStream -> inputStream.md5() }
    hashFile: (File) -> String? // Pass { file -> file.md5() }
) {
  // common logic to add property and compute hash goes here
}
```